### PR TITLE
Add PWA support

### LINKED ISSUE #11: Add Progressive Web App (PWA) support
Docusaurus has a [PWA plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-pwa) which adds a service worker in production builds, and can allow the app to be installed and run offline.  Let's add this so users can work with the site offline.

### PR DESCRIPTION
**Pull request by Kevan-Y**
Fixes #11.

Add PWA support.
image

### CODE REVIEW

**Comment by Kevan-Y** [comment]
@humphd I'm not sure how I suppose to resolve the conflict of `yarn.lock` since it is an autogenerated file. Edited might affect some stuff.

**Comment by humphd** [comment]
Also, you need to rebase this on current `main`, since we landed your Docusaurus version update already.

**Comment by humphd** [comment]
It also looks like we need better icons for the manifest, that one looks terrible blown-up.

**Comment by Kevan-Y** [comment]
> It also looks like we need better icons for the manifest, that one looks terrible blown-up.

Yeah that what I wanted to ask if it is possible to provide better icons.

File: docusaurus.config.js [hunk 1]
```diff
@@ -3,58 +3,114 @@ const darkCodeTheme = require('prism-react-renderer/themes/dracula');
 
 /** @type {import('@docusaurus/types').DocusaurusConfig} */
 module.exports = {
```
Comment (humphd) [inline docusaurus.config.js:5]: Why all the changes to this file?

**Review by humphd — CHANGES_REQUESTED** [review]

File: docusaurus.config.js (hunk 1 above)
Comment (Kevan-Y) [reply docusaurus.config.js:5]: I think it's my own VScode prettier formatter that changed this file. 

Maybe we should set up a prettier file to keep the same format everywhere?

Comment (humphd) [reply docusaurus.config.js:5]: We could file an issue to do that if you want to file another issue, but in this PR, please revert unnecessary styling changes.

**Review by humphd — CHANGES_REQUESTED** [review]
Let's figure out a solution to this icon.  I suspect you'll have to source it on the Seneca site, and provide different sizes.

**Comment by Kevan-Y** [comment]
@andrewillomitzer will also help review this.

**Comment by AndreWillomitzer** [comment]
### Code Review (In Progress):
1. As David mentioned and Kevan knows, the image in `static/manifest.json` needs to be higher resolution. I believe the issue with the blurring is because the image is 192px. Maybe there is an Adobe tool that will let you do it [Adobe Article](https://www.adobe.com/ca/creativecloud/photography/discover/increase-resolution.html)
2. In the `manifest.json` file, would it be possible to change the **description** to match the **name** property IPC144 - Course Notes?

Aside from that it looks great!!! Good job.

**Comment by Kevan-Y** [comment]
@humphd Did not find this current icon in higher resolution. 
I tried some solution @AndreWillomitzer provided and some online upscaling app, but it's doesn't look good.

There is this [icon](https://www.senecacollege.ca/student-services-and-support/mobile/_jcr_content/root/responsivegrid/container/container2/content/column_control/parsys_1/column_control_73692/parsys_1/image.img.png/1545199065914.png) that has better resolution but it's for the mobile app

**Comment by humphd** [comment]
I think we can file a follow-up issue to solve this.  I don't have an answer.  @Kevan-Y do you want to do that?

**Comment by Kevan-Y** [comment]
> I think we can file a follow-up issue to solve this. I don't have an answer. @Kevan-Y do you want to do that?

That works for me.

**Review by humphd — APPROVED** [review]

**Pull request merged**